internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -343,11 +343,22 @@ func readDirs(dir string) ([]string, error) {
 	return dirs, nil
 }
 
-func copyDir(from, to string) error {
+func copyDir(l *logger.Logger, from, to string, ignoreErrs bool) error {
+	var onErr func(error) error
+
+	if ignoreErrs {
+		onErr = func(err error) error {
+			// ignore all errors, just log them
+			l.Debugf("ignoring error: failed to copy %q to %q: %s", from, to, err.Error())
```

Comment:
I would consider making this an Info log, neither upgrades or errors are going to happen frequently enough for log spam to be an issue.

In general I would bias towards logging at the info level in the upgrade process so we aren't stuck trying to get rare failures to happen a second time with the log level changed to debug.

Reply:
Could agree more, analysing debug logs is a pain for the upgrade process 😄 
Thanks @michalpristas for this btw